### ISSUE TITLE
about Waterloo Exploration Database(WED)

### ISSUE THREAD

**Issue by chensming** [va1shn9v/PromptIR#4]
Thanks for the great research on multi degradation restoration. But when I tried to train the networks, I found the link of Waterloo Exploration Database did not work (http://ivc.uwaterloo.ca/database/WaterlooExploration/exploration_database_and_code.rar). 
Could you submit it or provide a download link? Thanks in advance.

**Comment by va1shn9v** [comment]
Thank you for taking an interest in our work. You can use this alternative link for downloading the wed dataset:  https://drive.google.com/file/d/19_mCE_GXfmE5yYsm-HEzuZQqmwMjPpJr/view?usp=sharing

**Comment by chensming** [comment]
Thanks for your answer! After downloading the datasets, I also want to know how to prepare the datasets in just the right way.
Here I have two questions:
1. how to prepare the images in the 'Train/denoise'?
2. How to place images in the directories BSD68 and Urban100?
![image](https://github.com/va1shn9v/PromptIR/assets/45591767/9171126d-2a06-4979-b7f1-2b3c23297d05)

**Comment by va1shn9v** [comment]
For Training data:
You should extract the images from zip files and just place the images directly in the directory. Both WED and BSD400 datasets are placed together in the Train/Denoise directory.

Similarly, in the testing data just unzip the images and but place them in their respective individual directories.

Yes, the gt images for deraining task are named norain-*.png and rainy are named as rain-*.png.

**Comment by chensming** [comment]
Recalling that “Finally, for image dehazing in the single-task setting, we utilize SOTS [31] dataset that contains 72,135 training images and 500 testing images”, I wonder do you mean the entire RESIDE Dataset? And how to divide them into a training set and a test set?

**Comment by va1shn9v** [comment]
For the dehazing task, we use the OTS(Outdoor training set) for training data, and SOTS(Synthetic objective Testing set) for the testing data. In this case, after extracting the downloaded files, directly place the extracted directories which would be named part1, part2,part3 and part4 in the Dehazing/synthetic directory.

**Comment by chensming** [comment]
> 

hi, when you evaluated on SOTS datasets, did you only use the indoor.rar? I found the number of images in the directory 'indoor/hazy' was 500. Here are the number of each directory:
```
- SOTS
+-- indoor.rar
+---- nyuhaze500
+-------- gt (50 images)
+-------- hazy (500 images)

+-- outdoor.zip
+---- outdoor
+-------- gt (492 images)
+-------- hazy (500 images)
```

**Comment by va1shn9v** [comment]
Hello, we evaluate the model on the outdoor split of SOTS dataset, i.e images from outdoor.zip file.

**Comment by chensming** [comment]
> Hello, we evaluate the model on the outdoor split of SOTS dataset, i.e images from outdoor.zip file.

so after unzipping outdoor.zip,  rename the directory 'outdoor/gt' to ‘dehaze/target',  and rename the directory 'outdoor/hazy' to 'dehaze/input'? (But I am worry about that the number of images in directory 'outdoor/gt' is 492 and that in directory 'outdoor/hazy' is 500, would that be a problem?)

**Comment by va1shn9v** [comment]
Yes, you would have to rename the directories as you have mentioned. 492 Images in the outdoor/gt directory is not an issue. Multiple test images can have the same groundtruth, as this is an synthetic dataset. There are a few test images, which only differ in the amount in haze in them. They would end up having the same groundtruth image, hence lower number of images in outdoor/gt, than outdoor/hazy.

**Comment by chensming** [comment]
> 

I see. What about the training dataset in the 'train/dehaze' directory? I have downloaded the OTS dataset. The directory is as followings:
```
+-- clear
+------ clear_images (8970 images, format: jpg)
+-- depth
+------ depth.zip.001 (8970 .mat files, format: mat)
+------ depth.zip.001 (4219 .mat files, format: mat)
+-- haze
+------ OTS.zip.001 (313950 images, format:.jpg)
+------ OTS.zip.002
+------  .......
+------ OTS.zip.010
+------ OTS.zip.011
```
![1688483296368](https://github.com/va1shn9v/PromptIR/assets/45591767/11cad0de-37ac-4dcf-9b84-415fedb0568c)

In the paper, "we utilize SOTS [31] dataset that contains 72,135 training images", could you give me an advice how can I get them(72135 images)? How to place them as you mention in INSTALL.md.I really have no idea.

```
└───Train
    ├───Dehaze
    │   ├───original
    │   └───synthetic
```

**Comment by va1shn9v** [comment]
Hi, kindly download the OTS split from this link : https://sites.google.com/view/reside-dehaze-datasets/reside-%CE%B2

If you open this link and navigate to dropbox download link, you would have to download 5 zip files. part1.zip, part2.zip,part3.zip,part4.zip and clear.zip.

After you expand the zip files you should, directly copy the part* directories into Dehaze/synthetic directory. Your directory structure would be linke Dehaze/synthetic/part1 , Dehaze/synthetic/part2 etc. The images present in the clear directory which is obtained by expanding clear.zip should be place in Dehaze/original directory.

You should have 2061 images in Dehaze/original directory and around 18200 images in each Dehaze/synthetic/part* directories.
The final directory structure would be like this:
<img width="207" alt="image" src="https://github.com/va1shn9v/PromptIR/assets/51985757/e968aa73-f353-49ae-b03c-7cd676377cde">

**Comment by chensming** [comment]
Hi, how can I get the bsd68? I didn't see the link in this project, so I went to another project and found bs68.
I follow the guide [KBNet/Denoising /README.md](https://github.com/zhangyi-3/KBNet/blob/main/Denoising/README.md#Gaussian-Image-Denoising). When I have downloaded it, I found it was gray image.
Here is the test003.jpg.

![test003](https://github.com/va1shn9v/PromptIR/assets/45591767/321ad1fa-8c3b-4b67-b6c0-59b97ec3e9e6)

But I found BSD400 is a color image. Here is the 2018.jpg.
![2018](https://github.com/va1shn9v/PromptIR/assets/45591767/e447319b-125d-463c-880e-988e500897f4)

So did I do something wrong?

**Comment by va1shn9v** [comment]
For BSD68 you should be using Color-BSD68 dataset. It is the same images but are not in grayscale. You can get it here : https://github.com/clausmichele/CBSD68-dataset/tree/master/CBSD68/original
